src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const currentState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
u can use destructuring for ur `action`, but up to u. For this case it not necessary)